opendevin/runtime/utils/image_agnostic.py

```diff
@@ -202,10 +213,25 @@ def get_od_sandbox_image(
     skip_init = False
     if image_exists:
         if is_eventstream_runtime:
-            skip_init = True
+            # An eventstream runtime image is already built for the base image (with poetry and dev dependencies)
+            # but it might not contain the latest version of the source code and dependencies.
+            # So we need to build a new (dev) image with the latest source code and dependencies.
+            # FIXME: In production, we should just build once (since the source code will not change)
             base_image = new_image_name
```

Comment:
I think for dev, we can just mount the project into the sandbox and manually launch it. Build a image maybe too heavy just for development.

Reply:
Good point! But beyond the codebase, we might still need to update the dependency - `mounting` & `poetry install` at the launch time might make the code a little bit complex, but can help reduce unnecessary re-build of docker images. And how to effectively switch between dev and prod (via config?) is also a question..

I'll give it more thoughts!